programs/ssl/ssl_client2.c

```diff
@@ -536,6 +551,45 @@ int idle( mbedtls_net_context *fd,
     return( 0 );
 }
 
+/* Unhexify `hex` into `dst`. `dst` must have
+ * size at least `strlen( hex ) / 2`. */
+int unhexify( char const *hex, unsigned char *dst )
+{
+    unsigned char c;
+    size_t j;
+    size_t len = strlen( hex );
+
+    if( len % 2 != 0 )
+        return( -1 );
+
+    for( j = 0; j < len; j += 2 )
+    {
+        c = hex[j];
+        if( c >= '0' && c <= '9' )
+            c -= '0';
+        else if( c >= 'a' && c <= 'f' )
+            c -= 'a' - 10;
```

Comment:
This is a bit confusing, I am not sure what is the desired output now, everything between 0xa and 0xf would be reduced to 0-5.

Reply:
@jarlamsa I'm not sure I follow - the logic is the same as `c = ( c - 'a' ) + 10`, which seems correct to me.

Reply:
Sorry, missed the double minus. 
But the refactoring comment is still valid although minor.

Reply:
@jarlamsa I agree, and we have considered reducing the overlap of `ssl_client2` and `ssl_server2` in other places, too. However, doing that is outside of the scope of this PR. I think @gilles-peskine-arm has done some work in that direction already (?).

Reply:
@hanno-arm You must be thinking of https://github.com/ARMmbed/mbedtls/pull/1272 where I proposed to introduce a `ssl_test_lib.c` used by both `ssl_client2` and `ssl_server2`. That work is currently stalled.